sqrl-tools/sqrl-config/src/main/java/com/datasqrl/config/ConnectorFactoryFactoryImpl.java

```diff
@@ -16,31 +16,53 @@
 public class ConnectorFactoryFactoryImpl implements ConnectorFactoryFactory {
 
   PackageJson packageJson;
+
+  private Optional<ConnectorConf> getConnectorConfig(String connectorName) {
+    Optional<EngineConfig> engineConfig = packageJson.getEngines().getEngineConfig("flink");
+    Preconditions.checkArgument(engineConfig.isPresent(), "Missing engine configuration for Flink");
+    ConnectorsConfig connectors = engineConfig.get().getConnectors();
+    return connectors.getConnectorConfig(connectorName);
+  }
+
+
+  @Override
+  public Optional<ConnectorFactory> create(SystemBuiltInConnectors builtInConnector) {
+
+    switch (builtInConnector) {
+      case PRINT_SINK:
+        return Optional.of(createPrintConnectorFactory(null));
+      case LOCAL_FILE_SOURCE:
+        return getConnectorConfig(builtInConnector.getName().getCanonical()).map(this::createLocalFile);
+      default:
+        throw new IllegalArgumentException("Unknown connector: " + builtInConnector);
+    }
+  }
+
   @Override
-  public Optional<ConnectorFactory> create(Type type, String name) {
+  public Optional<ConnectorFactory> create(Type engineType, String connectorName) {
+
+    // from conflict, include into the new logic
```

Comment:
?

Reply:
Prior to both PRs, this section was somewhat disorganized, allowing us to create connectors based on names and engineType. The logging PR introduced some refactoring, making engineType the sole determinant for selecting a factory. However, in the snowflake PR, you used the connectorName to decide whether to create an iceberg connector. I considered whether we could use the engineType for this purpose, but I'm not certain. Given the workaround comment, I decided it wasn't worth investigating at this point. I merely wanted to highlight that the inconsistency in the logic arose from a conflict, resulting in a state that might benefit from further refactoring.